src/Validator/Authentication.php

```diff
@@ -89,18 +98,31 @@ public function __construct($options = null)
         }
 
         if (is_array($options)) {
-            if (array_key_exists('adapter', $options)) {
+            if (isset($options['adapter'])) {
                 $this->setAdapter($options['adapter']);
             }
-            if (array_key_exists('identity', $options)) {
+            if (isset($options['identity'])) {
                 $this->setIdentity($options['identity']);
             }
-            if (array_key_exists('credential', $options)) {
+            if (isset($options['credential'])) {
                 $this->setCredential($options['credential']);
             }
-            if (array_key_exists('service', $options)) {
+            if (isset($options['service'])) {
                 $this->setService($options['service']);
             }
+            if (isset($options['code_map'])) {
+                foreach ($options['code_map'] as $code => $template) {
+                    if (empty($template) || ! is_string($template)) {
+                        throw new Exception\InvalidArgumentException(
+                            'Message key in code_map option must be a non-empty string'
+                        );
+                    }
+                    if (! isset($this->messageTemplates[$template])) {
+                        $this->messageTemplates[$template] = $this->messageTemplates[static::GENERAL];
+                    }
+                    $this->codeMap[(int) $code] = $template;
```

Comment:
Just a general question: why are the codes restricted to integers? I know we've run into problems in zend-db around this, as some extensions throw exceptions that use non-integer codes.

Reply:
I am not sure what was the reason for it, but it is what Result uses: https://github.com/zendframework/zend-authentication/blob/cbfec9c0270b8c3139c16d00ef93fc800d0dd190/src/Result.php#L76

Reply:
That's rationale enough for me - in fact, that may have been due to us running into those non-standard codes. I just wanted to make sure that there was a solid reason for it.

It might be good to indicate in the docs section that the code map maps _result codes_ to validation error codes. I wasn't clear on that, and this whole question could have gone away if it were documented. :)